data_safe_haven/pulumi/dynamic/remote_powershell.py

```diff
@@ -50,13 +50,8 @@ def create(self, props: dict[str, Any]) -> CreateResult:
             outs=outs,
         )
 
-    def delete(self, id_: str, props: dict[str, Any]) -> None:
-        """The Python SDK does not support configuration deletion"""
-        return
```

Comment:
We should include at least a comment explaining why `delete` is not needed here (probably because you can't delete the fact that a script has run). Also check whether `DshResourceProvider` requires this method to exist or not.

Reply:
I wasn't sure exactly the best thing to do here.

- It was clear that delete was not valid
- The parent class `DshResourceProvider` is not an `ABC` and the `delete` method is not an `abstractmethod`. So there is no error if it is not defined when creating a child of `DshResourceProvider`. ([see here](https://docs.python.org/3/library/abc.html?highlight=abc))

Having a closer look though, `DshResourceProvider`'s `update` method does call `delete`

https://github.com/alan-turing-institute/data-safe-haven/blob/4058ee0b00c168da98ede70ebaeadf36b457696e/data_safe_haven/pulumi/dynamic/dsh_resource_provider.py#L67-L76

I can't find this method every being called though.

So, maybe we should do one of

1. Redefine `update` in `RemoteScriptProvider` (seems like not a great idea).
2. Make the appropriate methods of `DshResourceProvider` abstract methods and add a comment to say what to do if `delete` is meaningless.

What do you think @jemrobinson?

In any case, I think this should be a new issue/PR.
I can revert the changes and ignore in CI.

Reply:
Captured in #1516